In [1]:
import pandas as pd
from pathlib import Path

In [2]:
path_lakes = Path('Z:/nahaUsers/casadje/EFASv6/lakes/tables')
old = pd.read_excel(path_lakes / 'efas6_lakes_20250218.xlsx')
new = pd.read_excel(path_lakes / 'efas6_lakes_20250304.xlsx')

print('{0} more lakes in the older version'.format(len(old) - len(new)))

31 more lakes in the older version


In [3]:
path_reservoirs = Path('Z:/nahaUsers/casadje/EFASv6/reservoirs/tables')
reservoirs = pd.read_excel(path_reservoirs / 'efas6_reservoirs_20250303+parameters.xlsx')

reservoirs.shape

(1359, 30)

In [4]:
# check lakes from EFAS 5 missing
set(old.loc[old.LakID.notnull(), 'LakID']).difference(new.LAK_ID)

set()

In [5]:
# check reservoirs from EFAS 5 missing
res_ids = set(old.loc[old.ResID.notnull(), 'ResID']).difference(new.RES_ID)
print(f'{len(res_ids)} reservoirs are missing')

32 reservoirs are missing


¿Están en la tabla de embalses?

In [6]:
res_ids.difference(reservoirs.RES_ID)

{126.0}

Eliminé ResID 126, así que hasta aquí todo bien.

In [7]:
# check lakes from GLWD_ID
set(old.loc[old.GLWD_ID.notnull(), 'GLWD_ID']).difference(new.GLWD_ID)

set()

In [33]:
# check lakes from HYLAK_ID
hylak_ids = set(old.loc[old.Hylak_id.notnull(), 'Hylak_id']).difference(new.HYLAK_ID)
print(f'{len(hylak_ids)} reservoirs are missing')

30 reservoirs are missing


¿Están en la tabla de embalses?

In [34]:
hylak_ids.difference(reservoirs.HYLAK_ID)

set()

Todo bien hasta aquí.